app/models/resource.rb

```diff
@@ -22,18 +22,18 @@ class Resource < ApplicationRecord
   validates :title, :resource_type, presence: true
 
   after_save do
-    SearchIndex.new(
-      model_name: self.class.name,
-      id: id,
-      async: true,
-    ).add
+    if SearchIndex.search_index_callbacks_enabled?
```

Comment:
This is closer to what we want, but still feels like there's some tight coupling between the `SearchIndex` class and the `AddToIndexJob`/`RemoveFromIndexJob` classes.

In my mind, I'd like to see a single direction of the flow

`Resource#after_save -> AddToIndexJob#perform_async -> SearchIndex#add`

So either `AddToIndexJob` or `SearchIndex` should handle checking `search_index_callbacks_enabled?` and deciding to log something or not. I'm leaning towards having `SearchIndex` make that decision since otherwise we'd have duplicate code in both `AddToIndexJob` and `RemoveFromIndexJob`, and they'd both be calling `SearchIndex#log_callback_warning` anyway.